### DISCUSSION TITLE
CPU Boost

### DISCUSSION THREAD

**Discussion by DevMore10** [seerge/g-helper#2736] [General]
Can Someone explain the Various CPU Boost Option in-Depth and what it actually does?

**Comment by Kikimahe** [comment]
Hi,

Take a look here #492

**Comment by DevMore10** [comment]
![image](https://github.com/seerge/g-helper/assets/136741231/eb17d96f-0358-41b3-b5d4-2017c0476213)

**Reply by DevMore10** [reply]
source: https://learn.microsoft.com/en-us/dotnet/api/microsoft.windows.eventtracing.power.processorperformanceboostmode?view=trace-processor-dotnet-1.0

**Reply by DevMore10** [reply]
i do not understand a thing it says

**Comment by DevMore10** [comment]
![image](https://github.com/seerge/g-helper/assets/136741231/bfab7127-eab3-49c0-b75e-2e12fc9257d1)

**Reply by DevMore10** [reply]
![image](https://github.com/seerge/g-helper/assets/136741231/21e82707-8631-471c-b460-a044fbaf3fb6)

**Comment by DevMore10** [comment]
https://learn.microsoft.com/en-us/windows-server/administration/performance-tuning/hardware/power/power-performance-tuning#processor-performance-boost-mode

**Comment by DevMore10** [comment]
![image](https://github.com/seerge/g-helper/assets/136741231/ee69eb9a-dbba-4f2c-b42c-64e1c1ed62d5)
![Uploading image.png…]()






























<html>
<body>
<p style="margin-bottom: 1rem; margin-top: 0px; color: rgb(242, 242, 242); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, &quot;Helvetica Neue&quot;, Arial, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(18, 18, 18); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">This drop-down is confusing, because Windows is mixing P-State and CPPC terminology.</p><p style="margin-bottom: 1rem; color: rgb(242, 242, 242); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, &quot;Helvetica Neue&quot;, Arial, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(18, 18, 18); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Here's the decoder ring:</p><h3 class="text-16 xs:text-18" style="margin: 1em 0px 0.5em; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: ; font-variant-east-asian: ; font-variant-alternates: ; font-variant-position: ; font-weight: ; font-stretch: ; font-size: 1.125rem; line-height: 1.5rem; font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, &quot;Helvetica Neue&quot;, Arial, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;, sans-serif; font-optical-sizing: ; font-kerning: ; font-feature-settings: ; font-variation-settings: ; color: rgb(242, 242, 242); letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(18, 18, 18); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">P-State:</h3><p style="margin-bottom: 1rem; color: rgb(242, 242, 242); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, &quot;Helvetica Neue&quot;, Arial, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(18, 18, 18); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">(P-State is the legacy performance management technology used if your system is configured to use BIOS rather than UEFI.)</p>
Setting | Behavior
-- | --

Disabled | No boost
Enabled | Boost is enabled
Efficient | Boost is enabled only if the Energy-Performance Bias (EPB) register is available.

<p style="margin-bottom: 1rem; color: rgb(242, 242, 242); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, &quot;Helvetica Neue&quot;, Arial, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(18, 18, 18); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Sources:</p><p style="margin-bottom: 1rem; color: rgb(242, 242, 242); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, &quot;Helvetica Neue&quot;, Arial, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(18, 18, 18); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><a rpl="" class="relative pointer-events-auto a
  
  
  
  
  hover:underline
  " href="https://docs.microsoft.com/en-us/windows-server/administration/performance-tuning/hardware/power/power-performance-tuning#processor-performance-boost-mode" rel="noopener nofollow ugc" target="_blank" style="color: var(--color-a-hover); text-decoration: underline; font-size: 1em; pointer-events: auto; position: relative; margin-top: 0px; margin-bottom: 0px;">Windows Server 2016 - Power and performance tuning</a></p><p style="margin-bottom: 0px; color: rgb(242, 242, 242); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, &quot;Helvetica Neue&quot;, Arial, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(18, 18, 18); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><a rpl="" class="relative pointer-events-auto a
  
  
  
  
  hover:underline
  " href="https://docs.microsoft.com/en-us/dotnet/api/microsoft.windows.eventtracing.power.processorperformanceboostmode?view=trace-processor-dotnet-1.0" rel="noopener nofollow ugc" target="_blank" style="color: var(--color-a-default); text-decoration: none; font-size: 1em; pointer-events: auto; position: relative; margin-top: 0px; margin-bottom: 0px;">ProcessorPerformanceBoostMode Enum</a></p>
</body>
</html>This drop-down is confusing, because Windows is mixing P-State and CPPC terminology.

Here's the decoder ring:

P-State:
(P-State is the legacy performance management technology used if your system is configured to use BIOS rather than UEFI.)

Setting	Behavior
Disabled	No boost
Enabled	Boost is enabled
Efficient	Boost is enabled only if the Energy-Performance Bias (EPB) register is available.
EPB was introduced in Sandy Bridge for consumer processors. I don't think AMD ever supported it, but I could be wrong.

CPPC:
(CPPC is the performance control mechanism used by modern machines.)

Setting	Behavior
Disabled	No boost
Efficient Enabled	Boost enabled; the OS specifies exactly how much boost it want
Aggressive	Boost enabled; always use maximum boost available
Aggressive At Guaranteed	Boost enabled; calculate amount of boost needed above guaranteed (i.e., all-core boost) frequency
Putting them together
Setting	P-State Behavior	CPPC Behavior
Disabled	No boost	No boost
Enabled	Boost enabled	Boost enabled; the OS specifies exactly how much boost it wants
Aggressive	Boost enabled	Boost enabled; always use maximum boost available
Efficient Enabled	Boost enabled only if the EPB register is available	Boost enabled; the OS specifies exactly how much boost it wants
Efficient Aggressive	Boost enabled only if the EPB register is available	Boost enabled; always use maximum boost available
Aggressive At Guaranteed	Boost enabled(?)	Boost enabled; calculate amount of boost needed above guaranteed (i.e., all-core boost) frequency
Efficient Aggressive At Guaranteed	Boost enabled only if the EPB register is available(?)	Boost enabled; calculate amount of boost needed above guaranteed (i.e., all-core boost) frequency
Sources:

[Windows Server 2016 - Power and performance tuning](https://docs.microsoft.com/en-us/windows-server/administration/performance-tuning/hardware/power/power-performance-tuning#processor-performance-boost-mode)

[ProcessorPerformanceBoostMode Enum](https://docs.microsoft.com/en-us/dotnet/api/microsoft.windows.eventtracing.power.processorperformanceboostmode?view=trace-processor-dotnet-1.0)
